packages/paste-website/src/component-examples/filter-group/components/DateTimePopover.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import {ErrorMessage} from '@hookform/error-message';
+import {useUID} from '@twilio-paste/uid-library';
+import {Box} from '@twilio-paste/box';
+import {Button} from '@twilio-paste/button';
+import {DatePicker} from '@twilio-paste/date-picker';
+import {Heading} from '@twilio-paste/heading';
+import {HelpText} from '@twilio-paste/help-text';
+import {Label} from '@twilio-paste/label';
+import {PopoverContainer, Popover, PopoverButton} from '@twilio-paste/popover';
+import {Text} from '@twilio-paste/text';
+import {TimePicker} from '@twilio-paste/time-picker';
+import {CalendarIcon} from '@twilio-paste/icons/esm/CalendarIcon';
+
+import type {DateTimePopoverProps} from '../types';
+
+export const DateTimePopover: React.FC<DateTimePopoverProps> = ({onApply, register, errors}) => {
+  const startId = `start-date-${useUID()}`;
+  const endId = `end-date-${useUID()}`;
+
+  return (
+    <PopoverContainer>
+      <PopoverButton variant="link" data-cy="custom-filter-group-popover-button">
+        <CalendarIcon decorative={false} title="Set custom date/time range" />
+      </PopoverButton>
+      <Popover aria-label="Custom date range" data-cy="custom-date-popover">
+        <Heading as="h2" variant="heading40">
+          Custom date range
+        </Heading>
+        <Text as="p" color="colorTextWeak" marginBottom="space70">
+          All dates/times in UTC. Usage data available up to 7 days.
+        </Text>
+        <Box display="flex" flexDirection="column" marginBottom="space50" rowGap="space50">
+          <Box display="flex" flexDirection="row" columnGap="space50">
+            <Box width="100%">
+              <Label htmlFor={`${startId}-date`} required>
+                Start date
+              </Label>
+              <DatePicker
+                id={`${startId}-date`}
+                aria-describedby={`${startId}-date-error`}
+                required
```

Comment:
thought (non-blocking): Does this accept min/max props? If so we can try passing max to startDate and min to endDate as better UX

Reply:
I'm gonna leave this as out of scope... I can add a ticket in the backlog to implement that.